src/Telemetry/Samples/CSharp/Program.cs

```diff
@@ -0,0 +1,209 @@
+                            SampleInt = 42,
+                            SampleDictionary = new()
+                            {
+                                { "key1", "value1" },
+                                { "key2", "value2" },
+                            },
+                            SampleGenericObject = new Dictionary<int, string>(),
+                            SampleGuid = Guid.NewGuid(),
+                            SampleException = CreateExceptionWithStackTrace(),
+                            SampleNullableWithValue = 123,
+                            SampleNullableWithNull = null,
+                        };
+                    TelemetryManager.LogObject(executionCompletedEvent);
+                }
+                catch (Exception exception)
+                {
+                    // RECOMMENDED: Log unhandled exceptions
+                    // Catch and log unhandled exceptions in main code.
+                    TelemetryManager.LogObject(exception);
+
+                    // Not recommended for production code.
+                    // This is only for debugging purposes of this sample program.
+                    Console.WriteLine($"UnhandledException: {exception}");
+                }
+                finally
+                {
+                    #if DEBUG
+                    if (TelemetryManager.Configuration.OutOfProcessUpload)
+                    {
+                        // we stop the stopwatch here because in debug mode
+                        // the teardown will await for the external process to finish
+                        // in order to keep listening for debug information
+                        stopwatch.Stop();
+                    }
+                    #endif
+                }
+            }
+
+            stopwatch.Stop();
+            Console.WriteLine($"Total time elapsed: {stopwatch.Elapsed}");
+            Environment.Exit(0);
+        }
+
+        private static Exception CreateExceptionWithStackTrace()
+        {
+            try
+            {
+                ThrowANestedException();
+                return null!;
+            }
+            catch (Exception exception)
+            {
+                return exception;
+            }
+        }
+
+        private static void ThrowANestedException() =>
+            ThrowAnException();
+
+        private static void ThrowAnException() =>
+            throw new System.IO.FileNotFoundException(@"File path 'C:\Users\johndoe\file.txt'");
+    }
+
+    public enum SampleEnumType
+    {
+        SampleEnumValue1,
+        SampleEnumValue2,
+    }
+
+    public record ExecutionCompleted
+    {
+        public DateTime SampleDateTime { get; set; }
+
+        public string? SampleString { get; set; }
+
+        public bool SampleBool { get; set; }
+
+        public SampleEnumType SampleEnum { get; set; }
+
+        [PiiData]
+        public string? SamplePII { get; set; }
+
+        [SerializeJson]

```

Comment:
is this still needed?

Reply:
If you mean the `[SerializeJson]` attribute, yes, it's needed to optionally serialize a whole object to json and send it as a telemetry payload to be consumed in Kusto.